std.posix.futimens() and related functions should take an optional pointer for "times"

### Zig Version

0.14.0-dev.2384+cbc05e0b1

### Steps to Reproduce and Observed Behavior

It should be possible to call `std.posix.futimens()` with a null pointer for `times`. This means that the current time must be used.

These related functions should also take an optional pointer for `times`:
* `c.futimes()`
* `c.futimens()`
* `c.utimes()`
* `c.utimensat()`

Example program:
```
const std = @import("std");

pub fn main() !void {
    const fd = try std.posix.open("test.txt", std.posix.O{ .CREAT = true, .ACCMODE = .WRONLY }, 0o644);
    defer std.posix.close(fd);

    try std.posix.futimens(fd, null);
}
```

The program does not build:
```
bug-report.zig:7:32: error: expected type '*const [2]os.linux.timespec__struct_1951', found '@TypeOf(null)'
    try std.posix.futimens(fd, null);
                               ^~~~
/usr/local/zig/lib/std/posix.zig:5760:34: note: parameter type declared here
pub fn futimens(fd: fd_t, times: *const [2]timespec) FutimensError!void {
                                 ^~~~~~~~~~~~~~~~~~
```

### Expected Behavior

The example program should build and run successfully.

--- a/lib/std/c.zig
+++ b/lib/std/c.zig
@@ -9387,11 +9387,11 @@ pub extern "c" fn malloc(usize) ?*anyopaque;
 pub extern "c" fn realloc(?*anyopaque, usize) ?*anyopaque;
 pub extern "c" fn free(?*anyopaque) void;
 
-pub extern "c" fn futimes(fd: fd_t, times: *[2]timeval) c_int;
-pub extern "c" fn utimes(path: [*:0]const u8, times: *[2]timeval) c_int;
+pub extern "c" fn futimes(fd: fd_t, times: ?*[2]timeval) c_int;
+pub extern "c" fn utimes(path: [*:0]const u8, times: ?*[2]timeval) c_int;
 
-pub extern "c" fn utimensat(dirfd: fd_t, pathname: [*:0]const u8, times: *[2]timespec, flags: u32) c_int;
-pub extern "c" fn futimens(fd: fd_t, times: *const [2]timespec) c_int;
+pub extern "c" fn utimensat(dirfd: fd_t, pathname: [*:0]const u8, times: ?*[2]timespec, flags: u32) c_int;
+pub extern "c" fn futimens(fd: fd_t, times: ?*const [2]timespec) c_int;
 
 pub extern "c" fn pthread_create(
     noalias newthread: *pthread_t,

--- a/lib/std/os/linux.zig
+++ b/lib/std/os/linux.zig
@@ -608,11 +608,11 @@ pub inline fn vfork() usize {
     return @call(.always_inline, syscall0, .{.vfork});
 }
 
-pub fn futimens(fd: i32, times: *const [2]timespec) usize {
+pub fn futimens(fd: i32, times: ?*const [2]timespec) usize {
     return utimensat(fd, null, times, 0);
 }
 
-pub fn utimensat(dirfd: i32, path: ?[*:0]const u8, times: *const [2]timespec, flags: u32) usize {
+pub fn utimensat(dirfd: i32, path: ?[*:0]const u8, times: ?*const [2]timespec, flags: u32) usize {
     return syscall4(.utimensat, @as(usize, @bitCast(@as(isize, dirfd))), @intFromPtr(path), @intFromPtr(times), flags);
 }
 

--- a/lib/std/posix.zig
+++ b/lib/std/posix.zig
@@ -5767,17 +5767,27 @@ pub const FutimensError = error{
     ReadOnlyFileSystem,
 } || UnexpectedError;
 
-pub fn futimens(fd: fd_t, times: *const [2]timespec) FutimensError!void {
+pub fn futimens(fd: fd_t, times: ?*const [2]timespec) FutimensError!void {
     if (native_os == .wasi and !builtin.link_libc) {
         // TODO WASI encodes `wasi.fstflags` to signify magic values
         // similar to UTIME_NOW and UTIME_OMIT. Currently, we ignore
         // this here, but we should really handle it somehow.
-        const atim = times[0].toTimestamp();
-        const mtim = times[1].toTimestamp();
-        switch (wasi.fd_filestat_set_times(fd, atim, mtim, .{
-            .ATIM = true,
-            .MTIM = true,
-        })) {
+        const error_code = blk: {
+            if (times) |times_arr| {
+                const atim = times_arr[0].toTimestamp();
+                const mtim = times_arr[1].toTimestamp();
+                break :blk wasi.fd_filestat_set_times(fd, atim, mtim, .{
+                    .ATIM = true,
+                    .MTIM = true,
+                });
+            }
+
+            break :blk wasi.fd_filestat_set_times(fd, 0, 0, .{
+                .ATIM_NOW = true,
+                .MTIM_NOW = true,
+            });
+        };
+        switch (error_code) {
             .SUCCESS => return,
             .ACCES => return error.AccessDenied,
             .PERM => return error.PermissionDenied,

